### ISSUE TITLE
wierd error message - "tread panicked, There to be a local ip"

### ISSUE THREAD

**Issue by KjellKriminell6** [thirtythreeforty/neolink#304]
hi!
Using command "neolink rtsp --config=your_config.yaml"

What does this error message mean?

**thread 'thread '<unnamed><unnamed>' panicked at '' panicked at 'There to be a local ip: PlatformNotSupported("windows")There to be a local ip: PlatformNotSupported("windows")', ', crates\core\src\bc_protocol\connection\udpconn\discover.rscrates\core\src\bc_protocol\connection\udpconn\discover.rs::304304::4848

note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace**

My config file looks like this:

_# A bind value of 0.0.0.0 means any network this computer can access
# You can change this to a specfic network e.g. "192.168.1.101" here
# Or to no networks e.g. this computer only "127.0.0.1"
bind = "0.0.0.0"

# Default port is 8554 but you can change it by uncommenting the following
# bind_port = 8554

# Uncomment the following and supply a path to a valid PEM
# to activate TLS encryption.
# The PEM should contain the certificate and the private key
# If TLS is activated you must connect with "rtsps://" and not "rtsp://"
# certificate = "/path/to/pem/with/cert/and/key"

# Choose if the client is required to provide a certificate signed by the server's CA.
# none|requested|required - default none
# tls_client_auth = "required"

# You can password protect the rtsp server mount points by adding users
# like the following me and someone. If you do not add [[users]]
# then anyone can connect without a password or username
# To access such a stream try using a url such as "rtsp://me:mepass@192.168.1.101/driveway"

# [[users]]
# name = "me"
# pass = "mepass"
#
# [[users]]
# name = "someone"
# pass = "someonepass"

[[cameras]]
name = "SAMENAMEASINREOLINKAPP"
username = "admin"
password = "SECRETPASSWORD"
uid = "MYUID"
# If you use a battery camera: **Instead** of an `address` supply the uid
# as follows
# uid = "ABCD01234567890EFG"

# By default any of the users can connect (or anyone at all if no users are specfied)
# You can uncomment the following to permit only specfic users
# permitted_users = [ "me" ]

# By default "both" "mainStream" and "subStream" are connected
# If your device has user connection limits try a single stream instead.
# stream = "mainStream"

# [[cameras]]
# name = "storage shed"
# username = "admin"
# password = "987654321"
# address = "192.168.1.245:9000"
# If you use a battery camera: **Instead** of an `address` supply the uid
# as follows
# adress = "192.168"

# If you use an NVR that relays several camera connections you can choose which
# camera to connect to with by setting the `channel_id`
#
# By default channel_id = 0. Eg the first connected camera on the device
# **Note**: that unlike in the offical client the  numbering starts from 0 not 1.
# An 8 channel NVR would have channels 0 through 7
# channel_id = 0_

**Comment by QuantumEntangledAndy** [comment]
So in order to use the remote UUID discorvery we need to send our IP address to the reolink as part of the handshake. For some reason the upstream crate [local-ip-address](https://github.com/EstebanBorai/local-ip-address) that we use for finding your IP is failing.

If you can read rust it is this bit of the code:

https://github.com/EstebanBorai/local-ip-address/blob/27886a95c33184052254666ba6ec02452ff7f954/src/lib.rs#L100-L105
```rust
let ip_addresses = crate::windows::list_local_ip_addresses(AF_INET)?;

ip_addresses
  .into_iter()
  .find(|ip_address| matches!(ip_address, IpAddr::V4(_)))
  .ok_or_else(|| Error::PlatformNotSupported(env::consts::OS.to_string()))
```

I suppose the obvious question is do you have a valid inet interface with an ipv4 address, or is there any other reason we might not be able to detect the IP address

**Comment by KjellKriminell6** [comment]
Thanks!

**Comment by PrplHaz4** [comment]
> So in order to use the remote UUID discorvery we need to send our IP address to the reolink as part of the handshake. For some reason the upstream crate [local-ip-address](https://github.com/EstebanBorai/local-ip-address) that we use for finding your IP is failing.
> 
> If you can read rust it is this bit of the code:
> 
> https://github.com/EstebanBorai/local-ip-address/blob/27886a95c33184052254666ba6ec02452ff7f954/src/lib.rs#L100-L105
> 
> ```rust
> let ip_addresses = crate::windows::list_local_ip_addresses(AF_INET)?;
> 
> ip_addresses
>   .into_iter()
>   .find(|ip_address| matches!(ip_address, IpAddr::V4(_)))
>   .ok_or_else(|| Error::PlatformNotSupported(env::consts::OS.to_string()))
> ```
> 
> I suppose the obvious question is do you have a valid inet interface with an ipv4 address, or is there any other reason we might not be able to detect the IP address

I'm guessing running neolink in a container might cause problems here if you're not using HOST networking?

**Comment by QuantumEntangledAndy** [comment]
Yes you cannot really use the UUID discovery in a container without using host mode (you'd have to open a ton of udp ports without host mode).

If your using UUID then use host mode or run it directly on the metal it's really not a big program and it's precompiled on so many things already.

**Comment by QuantumEntangledAndy** [comment]
If you have a camera that supports TCP connections (not the cellular or battery ones) then you can use the container anyway you want because there's only one tcp port and we open it up as part of our docker image.
